### ISSUE TITLE
Sparse Matrices and Linear SVM

### ISSUE THREAD

**Issue by crjackso** [accord-net/framework#212]
Hello,

I'd like to run a linear SVM classifier for text classification.  However, I'm having trouble seeing how this can be done in Accord with sparse matrices.  Because I'll be dealing with large feature sets (# of n grams), it's important that I use sparse matrices in the various computations.  For example, if I have 100,000 documents that I'd like to classify, I may have a training set matrix of 100,000 x 250,000, assuming there are 250,000 unique terms in the document set (I'm using TF-IDF).  Without representing this matrix in a sparse format, I may be looking at several gigs of memory consumption, which doesn't work.

The constructor for LinearCoordinateDescent requires a jagged array, not a sparse matrix.  Is what I'm trying to do possible?  I've included a rough example of what I'm trying to do below.  Thanks for your help!

```
var samples = new SparseMatrix(...); // from Math.NET...
var machine = new SupportVectorMachine(100);
var classifications = new double[...];
/// would like to pass in the sparse matrix below (or something else that's memory-friendly)
var teacher = new LinearCoordinateDescent(machine, samples.ToRowArrays(), classifications);
teacher.Run(true);
```

**Comment by cesarsouza** [comment]
Hello,

Thanks for bringing this issue. I have added support for sparse kernels in linear classification algorithms and it will be available in the next release, which I hope to finish in the next days.

Regards,
Cesar

**Comment by cesarsouza** [comment]
Fixed on release 3.3.0.
